Lib/test/test_importlib/test_util.py

```diff
@@ -775,5 +775,33 @@ def test_complete_multi_phase_init_module(self):
             self.run_with_own_gil(script)
 
 
+class MiscTests(unittest.TestCase):
+    def test_atomic_write_should_notice_incomplete_writes(self):
+        from importlib import _bootstrap_external
+        from test.support import os_helper
+        import _pyio
+        import os
+
+        oldwrite = os.write
+        seen_write = False
+
+        # emulate an os.write that only writes partial data
+        def write(fd, data):
+            nonlocal seen_write
+            seen_write = True
+            return oldwrite(fd, data[:100])
+
+        # need to patch _io to be _pyio, so that io.FileIO is affected by the
+        # os.write patch
+        with (unittest.mock.patch('importlib._bootstrap_external._io', _pyio),
+              unittest.mock.patch('os.write', write)):
```

Comment:
You don't have to go as far as mocking if you're just swapping out an attribute:

https://github.com/python/cpython/blob/03924b5deeb766fabd53ced28ba707e4dd08fb60/Lib/test/support/__init__.py#L1363-L1392

Reply:
Thanks, I indeed didn't know about it.